### DISCUSSION TITLE
Request Hook Invoked for All Query/Mutations

### DISCUSSION THREAD

**Discussion by browniefed** [hasura/graphql-engine#7413] [General]
## Why

For certain industries it is beneficial / required to track all data requests into the system. 

## Issue

Currently if a consumer of Hasura wants to track / log every request coming through the system there is no clean way of doing so. 

The only current solution is to leverage the AUTH_HOOK however if a user can't or is using a different authentication flow it becomes impossible to track all queries.

Tracking mutations can happen via event triggers but this does not solve the query tracking issue.

Other solutions would also include proxying all requests and logging, but that is also less than ideal as it requires more infrastructure.

## Request

A new environment variable that is an HTTP endpoint that gets invoked with every request/mutation but is non-blocking.

A non blocking endpoint that is invoked on every request whether it be a query/mutation, but does not block the flow of the query/mutation like AUTH_HOOK.

**Comment by motleydev** [comment]
This feature doesn't currently work this way, but if the table selector of event triggers behaved in a way that it would trigger the event on all tables and ll columns simulate the behavior you want?

**Reply by motleydev** [reply]
@tirumaraiselvan

**Reply by browniefed** [reply]
@motleydev I assumed AUTH_HOOK was invoked per-request no? Maybe I am mistaken and it doesn't receive a POST of the query/mutation +variables.

I am mostly requesting some way for us to log every request into the system. Whether it's a query, mutation, or initial subscription, and changes in `where` clauses if a subscription changes.

At the moment an HTTP Post that gets invoked seems easiest.



We need some way to log that our Doctors and the data they are requesting, and logging it. Then looking for irregularities, and or if someone is over-accessing information and then triggering alerts. Whether that is piping into Prometheus w/ stuff, or something else. We haven't decided on a system yet.
